dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -42,6 +42,41 @@ def login
     redirect_to auth_redirect_url.to_s
   end
 
+  def authenticate
+    id_token = params[:id_token]
+    return unauthorized_status unless id_token
+    begin
+      decoded_jwt_no_auth = JSON::JWT.decode(id_token, :skip_verification)
+    rescue
+      return unauthorized_status
+    end
+    # client_id is the aud[ience] in the JWT
+    extracted_client_id = decoded_jwt_no_auth[:aud]
+    extracted_issuer_id = decoded_jwt_no_auth[:iss]
+
+    integration = LtiIntegration.find_by({client_id: extracted_client_id, issuer: extracted_issuer_id})
+    return unauthorized_status unless integration
+
+    # verify the jwt via the integration's public keyset
+    decoded_jwt = get_decoded_jwt(integration, id_token)
```

Comment:
This raises an error if the jwt's signature isn't valid with the jwk_set, right? Should this be wrapped in a begin/rescue block?

Reply:
just added that!